### PR TITLE
Updates Fronius docs adding consumption path

### PR DESCRIPTION
**Pull request by luixal**
## Proposed change
Documentation improvement.

Adds config steps for Froinus meter installed in consumption path. I've reproduced and tested myself this configuration from 
https://github.com/home-assistant/home-assistant.io/issues/21804

Documentation gives credit to real author (not me).

## Type of change

Documentation

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [X] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

Doesn't edit previous content.

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: documents solution in closed issue #21804 

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @luixal, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#22320).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: source/_integrations/fronius.markdown [hunk 1]
```diff
@@ -78,6 +78,16 @@ When a device is not responding correctly the update interval will increase to 1
 
 ## Energy dashboard
 
+Home Assistant's Energy Dashboard expects some sources of data to be configured for it to work properly. They mainly are the energy you get from the grid, the energy you send to the grid, the energy produced by your solar panels and others relative to batteries and devices you can see below.
+
+The meter integrated with de Fronius can be installed (and configured) in two different positions: _"feed in path"_ (`meter_location` = 0) or _"in consumption path"_ (`meter_location` = 1). In the first case, the meter provides all data you need but, in the second case, it doesn't so you need to calculate some of them by creating some entities in your configuration file.
```
Comment (farmio) [inline source/_integrations/fronius.markdown]:
```suggestion
The meter integrated with de Fronius can be installed (and configured) in two different positions: _"feed in path"_ (`meter_location` = 0) or _"in consumption path"_ (`meter_location` = 1). In the first case, the meter provides all data you need but, in the second case, it doesn't so you need to calculate some of them by using helpers.
```

**Review by farmio — COMMENTED** [review]
Helper sensors (like integral) can now not only be configured in yaml so we should not be too specific about that in the Fronius documentation.

You can relatively link to different integrations like
`[Integration - Riemann sum integral](/integrations/integration/)` or
`[State-based template sensors](/integrations/template/#state-based-template-binary-sensors-buttons-numbers-selects-and-sensors)` (although I guess this heading may change one day so it might be better to just link to /template/ )

**Comment by luixal** [comment]
Not sure about which changes would you like me to do. It's my first time commiting to HA docs. I just placed the steps I followed to get this working :)

**Comment by farmio** [comment]
I just think we should not use exact yaml configuration of other integrations in the Fronius documentation page. When other integrations change some details once, we would have to notice and update this page too - just like it happend in 2022.4: Riemann integral can now be configured from UI instead of yaml. Maybe yaml config for Riemann gets deprecated at some point in the future - then the Fronius documentation is outdated as well. 

So maybe try to provide an overview of how this should be done instead of a copy&paste-yaml.

You did acutally add such a high-level description: 
> If a battery is connected to an inverter: Use [Riemann sum](/integrations/integration/) over `Power photovoltaics` entity (from power_flow endpoint found in your `SolarNet` device)

I'd describe the power/energy sold/bought things just like this.

**Comment by luixal** [comment]
Sorry, I just copy&paste the description you're pointing too from the same page (from the _feed in path_ original content), I didn't write that.

Thing is, the kind of explanation you want is out of my knowledge, I just wanted to "make it work" and put things together from the forum thread and the [github issue](https://github.com/home-assistant/home-assistant.io/issues/21804) you pointed me to and another user made it work.

It just feel better to publish this config, which is what I want "as a user" instead of keep it for myself. If you need a different documentation I think I won't have the time to dig deeper and understand it enough to write it properly.

File: source/_integrations/fronius.markdown [hunk 2]
```diff
@@ -88,6 +98,58 @@ Recommended energy dashboard configuration for meter location in feed in path (`
...
+
+```yaml
+template:
+  - sensor:
+    # Positive values are from grid
+    - name: "power_bought"
+      unit_of_measurement: W
+      state: >-
+        {{ max(states('sensor.power_grid_fronius_power_flow_0_XXX_XXX_XXX_XXX') | float, 0) }}
+      device_class: power
```
Comment (frenck) [inline source/_integrations/fronius.markdown]: This is a bug IMHO and not a documentable solution.

It is wrapping an existing sensor to solve an issue (as it seems).

File: source/_integrations/fronius.markdown [hunk 3]
```diff
@@ -88,6 +98,58 @@ Recommended energy dashboard configuration for meter location in feed in path (`
...
+  - platform: integration
+    source: sensor.power_sold
+    name: energy_sold
+    round: 2
+    
+  - platform: integration
+    source: sensor.power_bought
+    name: energy_bought
+    round: 2
+```
```
Comment (frenck) [inline source/_integrations/fronius.markdown]: This is not related to this integration and does not belong in this documentation. Instead you could write a note and link to the integration.

File: source/_integrations/fronius.markdown [hunk 4]
```diff
@@ -88,6 +98,58 @@ Recommended energy dashboard configuration for meter location in feed in path (`
...
+
+Now, you only need to configure Energy dashboard's params using the UI (first to values use the newly created entities, other ones are just the same as in the _"feed in path"_ config):
+
+- For `Grid consumption` use the meters `energy_bough` entity.
```
Comment (frenck) [inline source/_integrations/fronius.markdown]: "Grid consumption" isn't code, and should thus not be code fenced (same applies to a couple of other values below).

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/fronius.markdown (hunk 2 above)
Comment (farmio) [reply source/_integrations/fronius.markdown]: The "Power Grid" entity as provided from Fronius API is positive on import and negative on export. Hence the `max(state, 0)` and `max(0- state, 0)` to split it up into two entities - import and export to be used in energy (after integration).

File: source/_integrations/fronius.markdown (hunk 3 above)
Comment (luixal) [reply source/_integrations/fronius.markdown]: These are needed as they are the ones you can select in the Energy Dashboard configuration. No idea why, but it is working that way.

File: source/_integrations/fronius.markdown (hunk 4 above)
Comment (luixal) [reply source/_integrations/fronius.markdown]: Ok, was just to give attention to the field names in the UI.

EDIT: in fact, I was just keeping consistence with previous content in this documentation. Not sure if I should edit it.

File: source/_integrations/fronius.markdown (hunk 3 above)
Comment (frenck) [reply source/_integrations/fronius.markdown]: Sure, but you can do a lot more with entities. It is not the goal of this page to teach about other integration (like the integration integration).

Thus, this does not belong on this page. Please remove it.

File: source/_integrations/fronius.markdown (hunk 4 above)
Comment (frenck) [reply source/_integrations/fronius.markdown]: > Not sure if I should edit it.

🤷 I was reviewing your changes; not the full document.

File: source/_integrations/fronius.markdown (hunk 3 above)
Comment (luixal) [reply source/_integrations/fronius.markdown]: I think our focus is different. I'm writting the docs that my future me, as a user, would like to find to configure it's solar installation in HA. This is a 100% needed part for it to work. If I remove this, I'll leave a documentation that doesn't fully allow the user to get it working.

What do you think? I remove this and then... add some text saying that this should be donde in order for it to work? or do I leave the average user (not interested in the technical parts) alone?

Comment (farmio) [reply source/_integrations/fronius.markdown]: I think you are missing that this document here should *only* be documenting the Fronius integration. 
This does *not* include the `energy` integration which has its own [documentation documents](https://www.home-assistant.io/docs/energy).
And also not Riemann sum and state based template sensors - which also have their own sites. 

It would just add maintenance dept to document all this multiple times.

Comment (luixal) [reply source/_integrations/fronius.markdown]: Then I got it wrong as the original document is based on Fronius integration but also related to configure the data you get from this integration in the energy integration. Using it with the energy integration is 1/3 of the document and is what I found missing here. In fact, the matter of the consumption path VS feed in grid is not touch until the energy integration section.

If this page souldn't include information/examples about the energy integration, the Riemann sum or template sensor, this PR for this problem in detail is totally senseless and it's best to close it.

**Comment by farmio** [comment]
I've updated this PR with a brief version of a consupmtion path meter instruction, without unrelated integration documentation, so it is not stale-closed.

**Review by frenck — APPROVED** [review]
Thanks, @luixal 👍
